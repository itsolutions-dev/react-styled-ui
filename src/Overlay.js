import styled, { css } from './styledComponent';

const Overlay = styled({
  component: 'div',
  style: css`
    background: rgba(0,0,0,0.5);
    height: 100%;
    width: 100%;
  `,
});

export default Overlay;
