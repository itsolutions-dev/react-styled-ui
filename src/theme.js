export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
