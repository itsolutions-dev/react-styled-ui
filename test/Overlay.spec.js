import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import Overlay from '../src/Overlay';

describe('Overlay', () => {
  test('should render an Overlay', () => {
    const component = ReactTestRenderer.create(<Overlay theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<Overlay theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have correct styles', () => {
    const component = ReactTestRenderer.create(<Overlay theme={theme} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
