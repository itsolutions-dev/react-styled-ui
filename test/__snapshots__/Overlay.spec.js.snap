// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should have correct styles 1`] = `
.c0 {
  background: rgba(0,0,0,0.5);
  height: 100%;
  width: 100%;
}

<div
  className="c0"
/>
`;
